packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2023,12 +2023,28 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     // as the pre-formatting value of the previous pass (repeat call).
     final bool textChanged = _value?.text != value?.text;
     final bool isRepeat = value == _lastFormattedUnmodifiedTextEditingValue;
+    final bool isRepeatText = value?.text == _lastFormattedUnmodifiedTextEditingValue?.text;
 
-    if (textChanged && widget.inputFormatters != null && widget.inputFormatters.isNotEmpty) {
+    // There's no need to format when the value start to composing and repeat composing.
```

Comment:
I think this is more clear, but let me know if it doesn't seem accurate.

```suggestion
    // There's no need to format when starting to compose or when continuing
    // an existing composition.
```